### PR TITLE
fix(Badge): prevents unwanted re-rendering

### PR DESCRIPTION
**Pull request by langz**
fixes https://github.com/dnbexperience/eufemia/issues/4047

Here's a [CSB](https://codesandbox.io/p/sandbox/boring-fog-n7vr2y?workspaceId=9eb653cb-4406-4b8f-bc96-714927cc7fcd), using the build from this PR, where Badge doesn't re-render.

### CODE REVIEW

File: packages/dnb-eufemia/src/components/badge/__tests__/Badge.test.tsx [hunk 1]
```diff
@@ -120,7 +120,7 @@ describe('Badge', () => {
       (attr) => attr.name
     )
 
-    expect(attributes).toEqual(['role', 'class', 'aria-label'])
+    expect(attributes).toEqual(['role', 'class', 'top', 'aria-label'])
```
Comment (langz) [inline packages/dnb-eufemia/src/components/badge/__tests__/Badge.test.tsx:123]:
```suggestion
    expect(attributes).toEqual(['role', 'class', 'aria-label'])
```

**Review by joakbjerk — APPROVED** [review]
Nice! Looks good from testing and comparing in CSB 🔥

File: packages/dnb-eufemia/src/components/badge/__tests__/Badge.test.tsx (hunk 1 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/badge/__tests__/Badge.test.tsx:123]: Will have to look into this, as I don't think we want top to be an attribute 😓

Comment (langz) [reply packages/dnb-eufemia/src/components/badge/__tests__/Badge.test.tsx]: Fixed 🏁

**Pull request merged**